src/Mvc/Mvc.Core/src/FileResult.cs

```diff
@@ -59,5 +65,39 @@ public string FileDownloadName
         /// Gets or sets the value that enables range processing for the <see cref="FileResult"/>.
         /// </summary>
         public bool EnableRangeProcessing { get; set; }
+
+        /// <inheritdoc />
+        Task IResult.ExecuteAsync(HttpContext httpContext)
+        {
+            if (httpContext == null)
+            {
+                throw new ArgumentNullException(nameof(httpContext));
+            }
+
+            var loggerFactory = httpContext.RequestServices.GetRequiredService<ILoggerFactory>();
+            var logger = loggerFactory.CreateLogger<FileResult>();
+
+            var (range, rangeLenth, serveBody) = FileResultExecutorBase.SetHeadersAndLog(
+                httpContext,
+                this,
+                null,
+                EnableRangeProcessing,
+                LastModified,
+                EntityTag,
+                logger);
+
+            if (!serveBody || (range != null && rangeLenth == 0))
+            {
+                // No body to return or incorrect range data
+                return Task.CompletedTask;
+            }
+
+            if (range != null)
+            {
+                logger.WritingRangeToBody();
+            }
+
+            return FileResultExecutorBase.WriteFileAsync(httpContext, httpContext.Response.Body, range, rangeLenth);
```

Comment:
I'm pretty sure that `FileResultExecutorBase.WriteFileAsync` expects a `fileStream` as the second argument. It can get the `Response.Body` from the `HttpContext`, but I don't think we have the actual `fileStream` in the `FileResult` base class.

Thanks for taking the time to help out and submit this PR! Unfortunately, this *does* conflict with #32647, and I think we're going to end up merging that first.

Reply:
Thanks @halter73 for your insights

Reply:
No problem. It's too bad we can't take both contributions.

I think it might be best to look for other `help-wanted` issues because it sounds like @barahonajm also looking to do [more of these](https://github.com/dotnet/aspnetcore/pull/32647#issuecomment-842801026).